src/components/generate/GeneratePlaylistStepper.tsx

```diff
@@ -6,81 +9,141 @@ import { Stepper, StepList, Step, StepContent } from "@components/ui/Stepper"
 import useTranslation from "next-translate/useTranslation"
 import Link from "next/link"
 import { useState } from "react"
+import { Form } from "@components/ui/Form"
 
 const STEPS_COUNT = 3
 
+const formSchema = z
+  .object({
+    playlistType: z.enum(["new", "existing"]),
+    name: z.string().optional(),
+    playlistSelected: z.string().optional(),
+    isPrivate: z.boolean(),
+    bands: z.array(z.number().min(1)).nonempty("At least one band is required"),
+  })
+  .superRefine((data, ctx) => {
+    if (data.playlistType === "new" && !data.name) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "Name is required for NEW role",
+        path: ["name"],
+      })
+    }
+
+    if (data.playlistType === "existing" && !data.playlistSelected) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "Playlist is required for EXISTING role",
```

Comment:
Same as in https://github.com/DanielMoraDC/festwrap-ui/pull/11/files#r1899344268